collector/taskrun.go

```diff
@@ -49,17 +58,64 @@ func (f *trStartTimeEventFilter) Update(e event.UpdateEvent) bool {
 	return false
 }
 
-func SetupTaskRunController(mgr ctrl.Manager) error {
-	reconciler := &ReconcileTaskRun{
+func SetupTaskRunScheduleDurationController(mgr ctrl.Manager) error {
+	reconciler := &ReconcileTaskRunScheduled{
 		client:        mgr.GetClient(),
 		scheme:        mgr.GetScheme(),
-		eventRecorder: mgr.GetEventRecorderFor("MetricExporterTaskRuns"),
-		trCollector:   NewTaskRunCollector(),
+		eventRecorder: mgr.GetEventRecorderFor("MetricExporterScheduledTaskRuns"),
+		trCollector:   NewTaskRunScheduledCollector(),
 	}
 	return ctrl.NewControllerManagedBy(mgr).For(&v1beta1.TaskRun{}).WithEventFilter(&trStartTimeEventFilter{}).Complete(reconciler)
 }
 
-func (r *ReconcileTaskRun) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+type trRunningConditionEventFilter struct {
+}
+
+func (f *trRunningConditionEventFilter) Create(event.CreateEvent) bool {
+	return false
+}
+
+func (f *trRunningConditionEventFilter) Generic(event.GenericEvent) bool {
+	return false
+}
+
+func (f *trRunningConditionEventFilter) Delete(event.DeleteEvent) bool {
+	return false
+}
+
+func (f *trRunningConditionEventFilter) Update(e event.UpdateEvent) bool {
+	//TODO remember, keep track of when pipeline-service and RHTAP starts moving from v1beta1 to v1
+	oldTR, okold := e.ObjectOld.(*v1beta1.TaskRun)
+	newTR, oknew := e.ObjectNew.(*v1beta1.TaskRun)
+	if okold && oknew {
+		oldCondition := oldTR.Status.GetCondition(apis.ConditionSucceeded)
+		newCondition := newTR.Status.GetCondition(apis.ConditionSucceeded)
+		// if the new tr is not running, return false
+		if newCondition == nil || newCondition.Status != corev1.ConditionUnknown || newCondition.Reason != v1beta1.TaskRunReasonRunning.String() {
```

Comment:
checking for running is incorrect here ... that resulted in overlapping with the kubelet related metrics